pkg/test/local/environment.go

```diff
@@ -15,38 +15,62 @@
 package local
 
 import (
+	"fmt"
 	"testing"
 
+	"istio.io/istio/pkg/test/dependency"
 	"istio.io/istio/pkg/test/environment"
 	"istio.io/istio/pkg/test/internal"
 )
 
 // Environment a local environment for testing. It implements environment.Interface, and also
 // hosts publicly accessible methods that are specific to local environment.
 type Environment struct {
-	ctx internal.TestContext
+	ctx *internal.TestContext
 }
 
 var _ environment.Interface = &Environment{}
-var _ Internal = &Environment{}
+var _ internal.Environment = &Environment{}
 
 // NewEnvironment returns a new instance of local environment.
-func NewEnvironment(ctx internal.TestContext) (*Environment, error) {
-	return &Environment{
-		ctx: ctx,
-	}, nil
+func NewEnvironment() (*Environment, error) {
+	return &Environment{}, nil
+}
+
+// Initialize the environment. This is called once during the lifetime of the suite.
+func (e *Environment) Initialize(ctx *internal.TestContext) error {
+	e.ctx = ctx
+	return nil
+}
+
+// InitializeDependency is called when a new dependency is encountered during test run.
+func (e *Environment) InitializeDependency(ctx *internal.TestContext, d dependency.Instance) (interface{}, error) {
+	switch d {
+	// TODO
+	default:
+		return nil, fmt.Errorf("unrecognized dependency: %v", d)
+	}
 }
 
 // Configure applies the given configuration to the mesh.
 func (e *Environment) Configure(t testing.TB, config string) {
-	// TODO
-	panic("Not yet implemented")
+	for _, d := range e.ctx.Tracker().All() {
+		if configurable, ok := d.(internal.Configurable); ok {
+			err := configurable.ApplyConfig(config)
+			if err != nil {
+				t.Fatalf("Error applying configuration to dependencies: %v", err)
+			}
+		}
+	}
 }
 
 // GetMixer returns a deployed Mixer instance in the environment.
 func (e *Environment) GetMixer() (environment.DeployedMixer, error) {
-	// TODO
-	panic("Not yet implemented")
+	s, err := e.get(dependency.Mixer)
+	if err != nil {
+		return nil, err
+	}
+	return s.(environment.DeployedMixer), nil
```

Comment:
Should we need to be mindful of this cast? Return an error if if fails?

Reply:
This cast failing would indicate a gross programming error on our part. Personally, I'd prefer to get a big bad crash, rather than an error.

Reply:
when it crashes, will the error that's logged make it obvious?  If so, that's fine.

Reply:
It should be pretty visible (based on my memory of getting this wrong when doing development).